src/core/src/EnvironmentOptions.C

```diff
@@ -264,4 +149,174 @@ std::ostream& operator<<(std::ostream& os, const EnvOptionsValues & obj)
   return os;
 }
 
+
+void
+EnvOptionsValues::set_defaults()
+{
+  m_help = UQ_ENV_HELP;
+  m_numSubEnvironments = UQ_ENV_NUM_SUB_ENVIRONMENTS_ODV;
+  m_subDisplayFileName = UQ_ENV_SUB_DISPLAY_FILE_NAME_ODV;
+  m_subDisplayAllowAll = UQ_ENV_SUB_DISPLAY_ALLOW_ALL_ODV;
+  m_subDisplayAllowInter0 = UQ_ENV_SUB_DISPLAY_ALLOW_INTER0_ODV;
+  //m_subDisplayAllowedSet  = ;
+  m_displayVerbosity = UQ_ENV_DISPLAY_VERBOSITY_ODV;
+  m_syncVerbosity = UQ_ENV_SYNC_VERBOSITY_ODV;
+  m_checkingLevel = UQ_ENV_CHECKING_LEVEL_ODV;
+  m_rngType = UQ_ENV_RNG_TYPE_ODV;
+  m_seed = UQ_ENV_SEED_ODV;
+  m_platformName = UQ_ENV_PLATFORM_NAME_ODV;
+  m_identifyingString = UQ_ENV_IDENTIFYING_STRING_ODV;
+  m_numDebugParams = UQ_ENV_NUM_DEBUG_PARAMS_ODV;
+  m_debugParams.assign(m_numDebugParams, 0.);
+}
+
+
+void
+EnvOptionsValues::set_prefix(const char* prefix)
+{
+  m_prefix = (std::string)(prefix) + "env_";
+
+  m_option_help = m_prefix + "help";
+  m_option_numSubEnvironments = m_prefix + "numSubEnvironments";
+  m_option_subDisplayFileName = m_prefix + "subDisplayFileName";
+  m_option_subDisplayAllowAll = m_prefix + "subDisplayAllowAll";
+  m_option_subDisplayAllowInter0 = m_prefix + "subDisplayAllowInter0";
+  m_option_subDisplayAllowedSet = m_prefix + "subDisplayAllowedSet";
+  m_option_displayVerbosity = m_prefix + "displayVerbosity";
+  m_option_syncVerbosity = m_prefix + "syncVerbosity";
+  m_option_checkingLevel = m_prefix + "checkingLevel";
+  m_option_rngType = m_prefix + "rngType";
+  m_option_seed = m_prefix + "seed";
+  m_option_platformName = m_prefix + "platformName";
+  m_option_identifyingString = m_prefix + "identifyingString";
+
+}
+
+
+void
+EnvOptionsValues::parse(const BaseEnvironment* env, const char* prefix)
+{
+  m_env = env;
+
+  if (m_env->optionsInputFileName().empty()) {
```

Comment:
Should parse be robust to the case of an empty input file string?  For now, I'm leaving the error in place.  Seems calling parse means you really want to parse, but I could also see just silently skipping the parse if nothing to do.

Reply:
Nah, keep the error.  Belt-and-suspenders.